pkg/database/postgresql.go

```diff
@@ -40,6 +40,14 @@ func (info PostgreSQLDatabaseInfo) GetWriteResourceSQL() string {
 		"ON CONFLICT(uuid) DO UPDATE SET name=$4, namespace=$5, resource_version=$6, cluster_deleted_ts=$7, data=$8"
 }
 
+func (info PostgreSQLDatabaseInfo) GetNamespacedResourcesLimitedSQL() string {
+	return "SELECT created_at, data FROM resource WHERE kind=$1 AND api_version=$2 AND namespace=$3 ORDER BY created_at LIMIT $4"
+}
+
+func (info PostgreSQLDatabaseInfo) GetNamespacedResourcesLimitedContinueSQL() string {
+	return "SELECT created_at, data FROM resource WHERE kind=$1 AND api_version=$2 AND namespace=$3 AND created_at > $4 ORDER BY created_at LIMIT $5"
+}
```

Comment:
I don't know how likely it is that two resources share the order key (`created_at` in this case) but in case this could happen, couldn't we miss some resources in the pagination?

Let's say we have resources 1, 2, 3 and 4 created at ts A, B, B and C respectively and we are paginating the content in pieces of size 2.
The first page will have elements: 1 and 2, but the second page could have just the element 4 because we are querying that `created_at`>B
  
If we want to go with this route my proposal would be to do the `ORDER BY` by combining `created_at` with another field that makes the resource unique (e.g., the `uuid`)

```suggestion
func (info PostgreSQLDatabaseInfo) GetNamespacedResourcesLimitedContinueSQL() string {
	return "SELECT created_at, data FROM resource WHERE kind=$1 AND api_version=$2 AND namespace=$3 AND created_at > $4 AND uuid > $5 ORDER BY created_at, uuid LIMIT $6"
}
```

Reply:
Yes, that could be possible if both have the same nanosecond of insertion. I will prepare 4 rows with the scenario you exposed here and check.

Reply:
I fixed the scenario and created a little experiment:

```
## Example where two share `created_date` to ensure its working properly
kubectl exec -n postgresql -it pods/kubearchive-1 -- psql kubearchive -c "DELETE FROM resource;"
cat experiment.txt | kubectl exec -n postgresql -i pods/kubearchive-1 -- psql kubearchive
curl -s --cacert ca.crt -H "Authorization: Bearer $(kubectl create token default)" "https://localhost:8081/apis/batch/v1/namespaces/default/jobs" | jq '.items[].metadata.name'
curl -s --cacert ca.crt -H "Authorization: Bearer $(kubectl create token default)" "https://localhost:8081/apis/batch/v1/namespaces/default/jobs?limit=2" | jq '.items[].metadata.name'
export CONTINUE_TOKEN=$(curl -s --cacert ca.crt -H "Authorization: Bearer $(kubectl create token default)" "https://localhost:8081/apis/batch/v1/namespaces/default/jobs?limit=2" | jq -r '.metadata.continue')
curl -s --cacert ca.crt -H "Authorization: Bearer $(kubectl create token default)" "https://localhost:8081/apis/batch/v1/namespaces/default/jobs?limit=2&continue=$CONTINUE_TOKEN" | jq '.items[].metadata.name'
```

[experiment.txt](https://github.com/user-attachments/files/17258455/experiment.txt)

Reply:
Nice! this works as expected :)